### PR TITLE
Make base class for most iterative rule tests

### PR DESCRIPTION
**Pull request by alandpost**
Factor out the pattern of creating and tearing down the RuleTester in
each of the iterative rule unit test classes.  Requested by kokosing.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        closeAllRuntimeException(tester);
+        tester = null;
+    }
+
+    protected RuleTester getRuleTester()
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java]: We could shorten this to `ruleTester()` or maybe just `tester()`

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule.test;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
+
+public class RuleTest
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java]: Make this class abstract

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{
+    private RuleTester tester;
+
+    @BeforeClass
+    public void setUp()
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java]: Make these methods final so they can't accidentally be overridden

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddIntermediateAggregations.java [hunk 4]
```diff
@@ -44,31 +42,16 @@
 import static com.facebook.presto.sql.planner.plan.ExchangeNode.Scope.REMOTE;
 import static com.facebook.presto.sql.planner.plan.ExchangeNode.Type.GATHER;
 import static com.facebook.presto.sql.planner.plan.ExchangeNode.Type.REPARTITION;
-import static io.airlift.testing.Closeables.closeAllRuntimeException;
 
 public class TestAddIntermediateAggregations
+    extends RuleTest
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddIntermediateAggregations.java]: The formatting is off for all of these. It should be 8 spaces

Comment (alandpost) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddIntermediateAggregations.java]: It would have been nice for the style check in the build to notice that.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java (hunk 1 above)
Comment (alandpost) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTest.java]: I was just trying to follow the conventions of the codebase.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddIntermediateAggregations.java (hunk 4 above)
Comment (alandpost) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddIntermediateAggregations.java]: Fixed

**Comment by alandpost** [comment]
I addressed electrum's comments aside from the name of the tester accessor.

**Comment by electrum** [comment]
It's fine (and preferable) to use a short name for something like this. We do that in many other places.

**Comment by kokosing** [comment]
I applied comments and merged manually. See https://github.com/prestodb/presto/commit/b8a1bf4d7ed52c6cdae109239842efd48a7b1e75

**Comment by kokosing** [comment]
@alandpost I also remove `Teradata:rule-test` branch
